android-auto-app/src/main/java/com/mapbox/maps/testapp/auto/car/MapSession.kt

```diff
@@ -7,83 +7,51 @@ import android.content.res.Configuration
 import androidx.car.app.Screen
 import androidx.car.app.ScreenManager
 import androidx.car.app.Session
-import com.mapbox.maps.EdgeInsets
-import com.mapbox.maps.MapSurface
-import com.mapbox.maps.Style
-import com.mapbox.maps.extension.androidauto.initMapSurface
-import com.mapbox.maps.extension.style.layers.generated.skyLayer
-import com.mapbox.maps.extension.style.layers.properties.generated.SkyType
-import com.mapbox.maps.extension.style.sources.generated.rasterDemSource
-import com.mapbox.maps.extension.style.style
-import com.mapbox.maps.extension.style.terrain.generated.terrain
-import com.mapbox.maps.plugin.locationcomponent.location
-import com.mapbox.maps.testapp.auto.R
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.extension.androidauto.MapboxCarMap
 
 /**
  * Session class for the Mapbox Map sample app for Android Auto.
  */
+@OptIn(MapboxExperimental::class)
 class MapSession : Session() {
-  private lateinit var mapSurface: MapSurface
-  private val carCameraController = CarCameraController()
+  private val carMapShowcase = CarMapShowcase()
+  private var mapboxCarMap: MapboxCarMap? = null
 
   override fun onCreateScreen(intent: Intent): Screen {
-    val mapScreen = MapScreen(carContext)
-    initMapSurface(scrollListener = carCameraController) { surface ->
-      mapSurface = surface
-      carCameraController.init(
-        mapSurface,
-        EdgeInsets(
-          carContext.resources.getDimensionPixelSize(R.dimen.map_padding).toDouble(),
-          0.0,
-          0.0,
-          0.0
-        )
-      )
-      mapScreen.setMapCameraController(carCameraController)
-      loadStyle(surface)
-      initLocationComponent(surface)
-    }
+    // The onCreate is guaranteed to be called before onCreateScreen. You can pass the
+    // mapboxCarMap to other screens. Each screen can register and unregister observers.
+    // This allows you to scope behaviors to sessions, screens, or events.
+    val mapScreen = MapScreen(mapboxCarMap!!)
```

Comment:
is there a way to bypass the mapboxCarMap? could we add the initialisation of MapboxCarMap inside the Screen instead of the session?

Reply:
Yeah, that's the approach I put in the `README.md`. 
``` kotlin
  override fun onCreateScreen(intent: Intent): Screen {
    val mapInitOptions = MapInitOptions(carContext)
    val mapboxCarMap = MapboxCarMap(mapInitOptions)
    return MapScreen(mapboxCarMap)
  }
```

The reason you want to store a cached version of it, is if you want to have `MapboxCarMapObserver`s that are scoped to the `Session`. In this example, the `CarMapShowcase` would apply to every screen.

Reply:
Probably we should refactor the android auto app as well, to match the README?

Reply:
Would like to show that you can handle the map in the session. This is how 1tap needs to work. The README gives a hello-world